docs/apiSpec.yml

```diff
@@ -0,0 +1,458 @@
+openapi: "3.1.1"
+info:
+  title: Degree Prerequisite Tracker
+  description: "For all your degree planning needs!"
+  version: "1.0"
+  contact:
+    email: degreeprerequisitetracker@gmail.com
+
+tags:
+  - name: Sign Up/In
+  - name: Users
+  - name: Plans
+  - name: Departments
+  - name: Programs
+  - name: Subjects
+  - name: Courses
+
+paths:
+  /signup:
+    servers:
+      - url: https://eciuedmxxubiurblwicm.supabase.co/auth/v1
+    post:
+      tags:
+        - Sign Up/In
+      summary: Create an account
+      description: Provide an email and password to create a permanent account, don't change Apikey. If you have a prexisting account please navigate to /token to generate access token.
+      parameters:
+        - name: Apikey
+          in: header
+          required: true
+          schema:
+            type: string
+            example: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImVjaXVlZG14eHViaXVyYmx3aWNtIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Mjk5Nzg4MzgsImV4cCI6MjA0NTU1NDgzOH0.4Yj5yja7Phb6880eMh46WyyNJSfaHNLaOhPvlNkzuHM
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                email:
+                  type: string
+                  example: example@gmail.com
+                password:
+                  type: string
+                  example: password
+      responses:
+        "200":
+          description: Copy the access token returned and paste into authorization located at top of page.
```

Comment:
We should add the failure responses here, like when the password doesnt meet the minimum requirements, or an account with the email already exists